<launch>
    <arg name="mav_name" default="hummingbird"/>
    <arg name="use_drag_coeff" default="true"/>
    <arg name="drag_coeff_type" default="LFS"/> <!-- LFS, NM -->
    <arg name="fit_file_num" default="4"/>
    <arg name="result_file_num" default="1"/>
    <arg name="cost_type" default="EXTERNAL"/>   <!-- obs_free needs cost_type to be EXTERNAL.  EXTERNAL, LINEAR_LS -->
    <arg name="need_collision_free" default="false"/>
    <arg name="experiment_time" default="10"/>
    <arg name="experiment_times" default="4"/>
    <arg name="plot" default="false"/>
    <arg name="use_prior" default="true"/>
    <arg name="use_jps" default="true"/>
    <arg name="poly_offset" default="0.00"/>


    <arg name="w_rate" default="0.1"/>
    <arg name="max_vel" default="5.0"/>
    <arg name="heading" default="true"/>

    <arg name="expriment" default="circle_speedup_stay"/>
    <arg name="start_x" default="10"/>
    <arg name="start_y" default="0"/>
    <arg name="start_z" default="10"/>

    <!-- <arg name="expriment" default="circle_vertical"/>
    <arg name="start_x" default="5"/>
    <arg name="start_y" default="0"/>
    <arg name="start_z" default="10"/> -->
    
    <!-- <arg name="expriment" default="hover"/>
    <arg name="start_x" default="0"/>
    <arg name="start_y" default="0"/>
    <arg name="start_z" default="1"/> -->

    <arg name="hover_x" default="10"/>
    <arg name="hover_y" default="0"/>
    <arg name="hover_z" default="1"/>

    <!-- circle_speedup
    circle_speedup_stay
    circle_vertical
    hover 
    lemniscate
    lemniscate_speedup -->
    <group if="$(arg need_collision_free)">
        <include file="$(find flight_corridor)/launch/flight_corridor.launch">
            <arg name="flight_corridor/goal_x" value="$(arg hover_x)"/>
            <arg name="flight_corridor/goal_y" value="$(arg hover_y)"/>
            <arg name="flight_corridor/goal_z" value="$(arg hover_z)"/>
            <!-- <arg name="flight_corridor/offset" value="$(arg poly_offset)"/> -->
            <arg name="flight_corridor/use_jps" value="$(arg use_jps)"/>
            <arg name="flight_corridor/use_prior" value="$(arg use_prior)"/>
        </include>
    </group>

    <node pkg="mpc_ros" type="mpc_node.py" name="mpc_node" output="screen">
        <param name="quad_name" value="$(arg mav_name)"/>
        <param name="with_drag" value="$(arg use_drag_coeff)"/>
        <param name="drag_coeff_type" value="$(arg drag_coeff_type)"/>
        <param name="fit_file_num" value="$(arg fit_file_num)"/>
        <param name="result_file_num" value="$(arg result_file_num)"/>
        <param name="expriment" value="$(arg expriment)"/>
        <param name="experiment_time" value="$(arg experiment_time)"/>
        <param name="experiment_times" value="$(arg experiment_times)"/>
        <param name="hover_x" value="$(arg hover_x)"/>
        <param name="hover_y" value="$(arg hover_y)"/>
        <param name="hover_z" value="$(arg hover_z)"/>
        <param name="start_x" value="$(arg start_x)"/>
        <param name="start_y" value="$(arg start_y)"/>
        <param name="start_z" value="$(arg start_z)"/>
        <param name="plot" value="$(arg plot)"/>
        <!-- <param name="cost_type" value="$(arg cost_type)"/> -->
        <param name="need_collision_free" value="$(arg need_collision_free)"/>
        <param name="use_prior" value="$(arg use_prior)"/>
        <param name="poly_offset" value="$(arg poly_offset)"/>
        <param name="max_vel" value="$(arg max_vel)"/>
        <param name="w_rate" value="$(arg w_rate)"/>
        <param name="heading" value="$(arg heading)"/>
    </node>

    
</launch>